### PR TITLE
Refactor to work with arm-ttk and iaas

### PR DESCRIPTION
**Pull request by zhengchang907**
# Changes

* Remove **eap-session-replication/target/**
* Remove **JBoss-EAP_on_Azure/target/**
* Refactor code structure for all offers
* Add pom.xml for all offers to work with azure-javaee-iaas-parent
* Add generateOffers.sh to run `mvn -Ptemplate-validation-tests clean install` for all offers
* Add validateOffers.sh to run `Test-AzTemplate` for all offers
* Update `validate-templates.yaml` to work with new structure
* Add _/target_ to .gitignore

# Test runs
https://github.com/Azure/rhel-jboss-templates/actions/runs/1821678081

# How to use

1. Checkout [Azure/arm-ttk](https://github.com/Azure/arm-ttk) at the same directory where _Azure/rhel-jboss-templates_ is located, side by side.
2. To validate all offers, run `./validateOffers.sh`. If you'd like to validate only one of those offers, for example **eap73-rhel8-payg**, run `./../arm-ttk/arm-ttk/Test-AzTemplate.sh -TemplatePath eap73-rhel8-payg/src/main/arm`
2. If you'd like to pack artifacts for all offers, run `./generateOffers.sh`. If you'd like to pack artifact for one of offers, for example **eap73-rhel8-payg**, run `mvn -Ptemplate-validation-tests clean install` under directory _eap73-rhel8-payg/_

# Questions remain

* I think we should avoid pushing _/target_ directories or .war files to GitHub?
* ~This change will break current pipeline **Validate Deployment Templates** because the code structure is changed~
* Should we add another pipeline to build all offers?

Signed-off-by: Zheng Chang <zhengchang@microsoft.com>

### CODE REVIEW

**Review by edburns — APPROVED** [review]
I will work with the Red Hat team to coordinate getting this merged.

File: eap73-rhel8-byos-multivm/pom.xml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        <groupId>com.microsoft.azure.iaas</groupId>
+        <artifactId>azure-javaee-iaas-parent</artifactId>
+        <version>1.0.13</version>
+        <relativePath></relativePath>
+    </parent>
+    <packaging>pom</packaging>
+    <name>${project.artifactId}</name>
+    <properties>
+        <artifactsLocationBase>https://raw.githubusercontent.com/oracle/${git.repo}/${git.tag}/rhel-jboss-templates</artifactsLocationBase>
+        <template.validation.tests.directory>${basedir}/../../../arm-ttk/arm-ttk</template.validation.tests.directory>
```
Comment (jamesfalkner) [inline eap73-rhel8-byos-multivm/pom.xml:20]: @edburns `oracle` ?

**Comment by jamesfalkner** [comment]
Can you make sure that the GitHub action in place currently continues to work  (and validates the templates on any pull request / push)? I noticed it failed. It's fine if the GitHub action does it in "the new way" (e.g. with Maven) as long as it works. cc @edburns

File: eap73-rhel8-byos-multivm/pom.xml (hunk 1 above)
Comment (zhengchang907) [reply eap73-rhel8-byos-multivm/pom.xml:20]: @jamesfalkner Hi James, it is an unused property, I will remove it along with the code change related to making pipeline works with "the new way"

**Review by jamesfalkner — APPROVED** [review]

**Pull request merged**